ably/realtime_conn.go

```diff
@@ -880,6 +880,33 @@ func (c *Connection) reauthorize(arg connArgs) {
 	c.reconnect(arg)
 }
 
+func (c *Connection) onClientAuthorize(ctx context.Context, token *TokenDetails) {
```

Comment:
This callbacks feels a bit misleading, it feels it gets called when client is authorised. Can we not make it callback and just change method name to `authorizeClient`. I believe thats what it does after receiving token. 
PS. if there is a name for callback, it should be `onClientTokenReceived`

Reply:
`onClientAuthorize` isn't a great name but `onClientTokenReceived` would be also misleading, because this is supposed to be called when the `Authorize` public method is explicitly called, not any time the client gets a token.

Renamed to `onExplicitAuthorize` at 730615d.

Reply:
I still think naming can be improved. I will approve it for now